server/middleware/registration/registrationService.test.js

```diff
@@ -0,0 +1,64 @@
+const registrationService = require('./index');
+const httpMocks = require('node-mocks-http');
+const mockMongoUtil = require('../../utils/mongoUtil');
+
+jest.mock('../../utils/mongoUtil');
+let collectionObj = mockMongoUtil.fetchCollection.mockImplementationOnce(jest.fn());
+collectionObj.insertOne = jest.fn();
+const errorMessage = {
+  data: 'There was a problem with your request. Please try again later.'
+}
+
+jest.mock('cryptr', () => {
+  const mockPlainText =
+    'mongodb://fakeUser:fakePassword@mongodb.fakeDomain.com:27017/fakeDb';
+  return jest.fn().mockImplementation(() => {
+    return { decrypt: () => mockPlainText };
+  });
+});
+
+let request, response;
+describe('Registration API', () => {
+  beforeEach(() => {
+    request = httpMocks.createRequest({
+      method: 'POST',
+      url: 'api/register',
+      body: {
+        firstName: 'Testing',
+        lastName: 'User',
+        prefLoc: 'NYC',
+        title: 'Software Engineer',
+        email: 'test@dev.com',
+        userName: 'devtest2021',
+        password: "H@llo2021!",
+      }
+    });
+  });
+
+  it('should add a new user document to collection ', async () => {
+    response = httpMocks.createResponse();
+    request.body.collectionObj = collectionObj;
```

Comment:
let's not do this hack; find a way to mock mongoUtil

Reply:
resolved in commit # https://github.com/cfernando415/femmecubator-api-repo/pull/36/commits/9829f4621df493aa4c36047ffdb6bbeeeeb87451